mmdet/models/dense_heads/corner_head.py

```diff
@@ -0,0 +1,1010 @@
+                br_heat (Tensor): Predicted bottom-right corner heatmap.
+                tl_emb (Tensor | None): Predicted top-left embedding heatmap.
+                    None for `self.with_corner_emb == False`.
+                br_emb (Tensor | None): Predicted bottom-right embedding
+                    heatmap. None for `self.with_corner_emb == False`.
+                tl_off (Tensor): Predicted top-left offset heatmap.
+                br_off (Tensor): Predicted bottom-right offset heatmap.
+                tl_pool (Tensor): Top-left corner pool feature. Not must have.
+                br_pool (Tensor): Bottom-right corner pool feature. Not must
+                    have.
+        """
+        tl_pool = self.tl_pool[lvl_ind](x)
+        tl_heat = self.tl_heat[lvl_ind](tl_pool)
+        br_pool = self.br_pool[lvl_ind](x)
+        br_heat = self.br_heat[lvl_ind](br_pool)
+
+        tl_emb, br_emb = None, None
+        if self.with_corner_emb:
+            tl_emb = self.tl_emb[lvl_ind](tl_pool)
+            br_emb = self.br_emb[lvl_ind](br_pool)
+
+        tl_off = self.tl_off[lvl_ind](tl_pool)
+        br_off = self.br_off[lvl_ind](br_pool)
+
+        result_list = [tl_heat, br_heat, tl_emb, br_emb, tl_off, br_off]
+        if return_pool:
+            result_list.append(tl_pool)
+            result_list.append(br_pool)
+
+        return result_list
+
+    def get_targets(self,
+                    gt_bboxes,
+                    gt_labels,
+                    feat_shape,
+                    img_shape,
+                    with_corner_emb=False,
+                    with_guiding_shift=False,
+                    with_centripetal_shift=False):
+        """Generate corner targets.
+
+        Including corner heatmap, corner offset.
+        Optional: corner embedding, corner guiding shift, centripetal shift.
+        For CornerNet, we generate corner heatmap, corner offset and corner
+            embedding from this function.
+        For CentripetalNet, we generate corner heatmap, corner offset, guiding
+            shift and centripetal shift from this function.
+
+        Args:
+            gt_bboxes (list[Tensor]): Ground truth bboxes of each image, each
+                has shape (num_gt, 4).
+            gt_labels (list[Tensor]): Ground truth labels of each box, each has
+                shape (num_gt,).
+            feat_shape (list[int]): Shape of output feature,
+                [batch, channel, height, width].
+            img_shape (list[int]): Shape of input image,
+                [height, width, channel].
+            with_corner_emb (bool): Generate corner embedding target or not.
+                Default: False.
+            with_guiding_shift (bool): Generate guiding shift target or not.
+                Default: False.
+            with_centripetal_shift (bool): Generate centripetal shift target or
+                not. Default: False.
+
+        Returns:
+            dict:
```

Comment:
Add an overall description after ':' like the following:

```python
dict: xxxxxx, containing the following keys:
    
    - topleft_heatmap (Tensor): xxxxxx
    - bottomright_heatmap (Tensor): xxxxx
 

```
Also check other places.